### DISCUSSION TITLE
Spring Kafka  using NonBlocking retries in 2.7.X version of Springboot.

### DISCUSSION THREAD

**Discussion by kasarlaharish1** [spring-projects/spring-kafka#2809] [Q&A] [answered]
Hi Team,

I have query regarding the Spring Kafka Non Blocking retries .
As mentioned in this link: https://docs.spring.io/spring-kafka/docs/2.8.11/reference/html/#retry-topic
 It says "This is an experimental feature and the usual rule of no breaking API changes does not apply to this feature until the experimental designation is removed. Users are encouraged to try out the feature and provide feedback via GitHub Issues or GitHub discussions. This is regarding the API only; the feature is considered to be complete, and robust." 

My question is did any one find any outstanding issues with using Nonblocking retries when using in spring kafka 2.8.x version?
Can we use these Nonblocking retries in the 2.8.11 version of spring kafka?
My requirement is to use in springboot 2.7 version only not in 3.X  version of springboot.

Thanks in advance.

**Comment by kasarlaharish1** [comment]
@garyrussell : Can you please comment on this.

Thanks.

**Reply by kasarlaharish1** [reply]
Hi @garyrussell , 
Can you please share your observations of having any major issue of using Non blocking retries of spring kafka 2.9.12 version with springboot 2.7.x version.

Thanks.

**Comment by garyrussell** [comment]
2.8.x is no longer supported as OSS (https://spring.io/projects/spring-kafka#support); use 2.9.12 or later with Boot 2.7.x.

See the matrix on the project page https://spring.io/projects/spring-kafka#overview

**Reply by kasarlaharish1** [reply]
Hi @garyrussell ,
Thanks for the response, In 2.9.12 version of spring kafka https://docs.spring.io/spring-kafka/docs/2.9.12/reference/html/#retry-topic 
also it is mentioned as Experimental  feature, can we go ahead and use it for prod? Just want to check if we have any outstanding issues related to this experimental feature for non blocking retries which is caused due to the springboot 2.7x vs springboot 3.x.

**Reply by garyrussell** [reply, marked as answer]
The API remained stable as of 2.9.x; the comment has been removed in the 3.0.x documentation. As the comment said, it was only the API that might change; the implementation has always been stable since the first release it appeared in.
